aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -957,18 +957,19 @@ TORCH_IMPL_FUNC(softmax_backward_cuda_out)
   host_softmax_backward<SoftMaxBackwardEpilogue, false>(tmp, output, dim, half_to_float, grad_input);
 }
 
-Tensor masked_softmax_cuda(const Tensor& input_, const Tensor& mask_, const c10::optional<int64_t> dim_) {
+Tensor masked_softmax_cuda(const Tensor& input_, const Tensor& mask_, const c10::optional<int64_t> dim_, const c10::optional<int64_t> mask_type_) {
   Tensor output = at::empty_like(input_, input_.options());
   TORCH_CHECK(mask_.scalar_type() == ScalarType::Bool, "Mask should be a boolean tensor");
+  TORCH_CHECK(mask_type_, "expected `mask_type` when `mask` supplied");
 
   // If input is [B, H, T, T] and mask is [B, T]
   // we have special fast kernel
-  bool is_BxT_mask = (input_.dim() == 4 && mask_.dim() == 2 && input_.size(0) == mask_.size(0) && input_.size(2) == mask_.size(1) && input_.size(3) == mask_.size(1));
+  bool is_BxT_mask = (mask_type_ == 2) && (input_.dim() == 4 && mask_.dim() == 2 && input_.size(0) == mask_.size(0) && input_.size(2) == mask_.size(1) && input_.size(3) == mask_.size(1));
```

Comment:
Can we use Enum instead of hardcoded number here? Or at least we need some comments indicate what mask_type==1 or ==2 means

Reply:
I was originally looking to use an Enum but couldn't figure out where the best location for it was? If you had a suggestion that would be appreciated and then I could go back and change the types.

Reply:
> 
Since this is being passed as param of function, I guess you can keep it as number. But please put some comments at the beginning of the function, to clarify each number's meaning.